### PR TITLE
Add Parallax2D progress report post

### PR DESCRIPTION
**Pull request by markdibarry**
Feel free to proofread or call out anything you see that should be changed.

### CODE REVIEW

File: collections/_article/parallax-progress-report.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+<video autoplay loop muted playsinline title="An demonstration of the parallax effect zoomed out">
+  <source src="/storage/blog/parallax2d/zoom_out.mp4" type="video/mp4">
+</video>
+
+Increasing the `repeat_times` will allow for the number of repeats to grow and spread outward, so that you can zoom out and not reveal the magic going on behind the scenes. Eventually, it'd be nice to have a flag to have this adjust automatically, but this is a good start!
+
+#### What else?
+
+`Parallax2D` is now compatible with `BackBufferCopy` and `CanvasGroup`, and you're also free to rotate the camera without breaking the parallax effect! Some things needed shifted around in the `ParallaxBackground` code to accommodate the new changes and I discovered a nice surprise. Through some basic tests with 20 complex layers (comparing before and after the feature was added), I saw a good performance boost with both the old system and the new one with the changes... which is nice!
```
Comment (clayjohn) [inline collections/_article/parallax-progress-report.md]:
```suggestion
`Parallax2D` is now compatible with `BackBufferCopy` and `CanvasGroup`, and you're also free to rotate the camera without breaking the parallax effect! Some things needed shifting around in the `ParallaxBackground` code to accommodate the new changes and I discovered a nice surprise. Through some basic tests with 20 complex layers (comparing before and after the feature was added), I saw a good performance boost with both the old system and the new one with the changes... which is nice!
```

**Review by clayjohn — COMMENTED** [review]
This was a very fun read, great job! 

I think users will want a clear statement on the plan for ParallaxLayer and ParallaxBackground.  In my mind the plan is not to touch them much and to focus on Parallax2D instead. In other words, we would accept bug fixes for them, but not new features and all new developments would be with Parallax2D. Does that align with your understanding as well?

Also, this post should explicitly say that deprecated does not mean removed. We will not be removing the old nodes, we will just strongly recommend that they aren't used. 

Finally we should say that the old nodes won't be deprecated until we are confident that they are fully superseded by Parallax2D. That could happen as early as 4.4, but might take a couple of releases.

File: collections/_article/parallax-progress-report.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+![An example of converting to Parallax2D](/storage/blog/parallax2d/button.webp)
+
+### Why the change?
+
+There've been a lot of proposals and issues regarding Godot's parallax system over the last few years. *Bogusly*, these had trouble getting movement without a significant overhaul of the two nodes that it's comprised of: `ParallaxBackground` and `ParallaxLayer`. Godot has introduced so many new features and upgrades in related areas (especially with 4.0) that the parallax system didn't originally need to support (because they didn't exist; A reasonable justification indeed!). I recently ran into one such case working on a faux-lighting system to replicate the look of lights on early video game consoles.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
There have been a lot of proposals and issues regarding Godot's parallax system over the last few years. These had trouble getting movement without a significant overhaul of the two nodes that it's comprised of: `ParallaxBackground` and `ParallaxLayer`. Godot has introduced so many new features and upgrades in related areas (especially with 4.0) that the parallax system didn't originally need to support (because they didn't exist; a reasonable justification indeed!). I recently ran into one such case working on a faux-lighting system to replicate the look of lights on early video game consoles.
```

Comment (JohnVeness) [reply collections/_article/parallax-progress-report.md]: "There've" is legitimate, but a very unusual contraction. Bogus==not genuine, so not sure how that applies.

By the way, what era are you considering "early video game consoles"?

File: collections/_article/parallax-progress-report.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+### Weren't you gonna talk about parallax?
+
+Right! Along the way to my destination of "artificial luminescence", I drove too fast and hit a speed bump. To have one object mask another, they need to be in the same `CanvasLayer`, which introduces a unique problem: `ParallaxLayer`s are easy to setup, but they're restricted to being direct children of a `ParallaxBackground`. Unfortunately (even more troublesome), `ParallaxBackground`'s base class is `CanvasLayer` - It's on an entirely different layer, making it completely incompatible and unseen when using a screen-reading shader. How inflexible! It looked like my only options were to give up or attempt to make `ParallaxBackground` somehow compatible with `BackBufferCopy` and `CanvasGroup`.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
Right! Along the way to my destination of "artificial luminescence", I drove too fast and hit a speed bump. To have one object mask another, they need to be in the same `CanvasLayer`, which introduces a unique problem: `ParallaxLayer`s are easy to setup, but they're restricted to being direct children of a `ParallaxBackground`. Unfortunately (even more troublesome), `ParallaxBackground`'s base class is `CanvasLayer` - it's on an entirely different layer, making it completely incompatible and unseen when using a screen-reading shader. How inflexible! It looked like my only options were to give up or attempt to make `ParallaxBackground` somehow compatible with `BackBufferCopy` and `CanvasGroup`.
```

File: collections/_article/parallax-progress-report.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+I initially, *innocently* assessed that this issue would be swiftly solved by changing `ParallaxBackground`'s base class from `CanvasLayer` to `Node2D`. It makes sense, right? `ParallaxBackground` doesn't really *act* like a `CanvasLayer`... so, instead of a bunch of workarounds to make a `CanvasLayer` behave like a `Node2D`, why not just start out on the right foot? But... that sounds too *easy!* It ate at me that `CanvasLayer` seemed like such a *specific* choice; I just had to know why!
+
+`ParallaxLayer` not only provides a parallax effect, but also an award-winning "infinite scrolling" bonus effect ("Best in Class" for 2018 AND 2019 - The class being `ParallaxLayer`). Of course, actual infinite scrolling isn't possible (right?), so, behind the scenes, Godot performs a bit of visual trickery:
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
`ParallaxLayer` not only provides a parallax effect, but also an award-winning "infinite scrolling" bonus effect ("Best in Class" for 2018 AND 2019 - the class being `ParallaxLayer`). Of course, actual infinite scrolling isn't possible (right?), so, behind the scenes, Godot performs a bit of visual trickery:
```
I don't really understand the best in class comment.

File: collections/_article/parallax-progress-report.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+<video autoplay loop muted playsinline title="Godot mascot riding in a convertible on a stage">
+  <source src="/storage/blog/parallax2d/secret.mp4" type="video/mp4">
+</video>
+
+It renders the children of a `ParallaxLayer` a second time, so that it can look like it's wrapping around, and when the screen scrolls far enough - *ZIP!* - it whips back like a typewriter (an ancient writing tool) and makes it *appear* like it's scrolling forever!
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
It renders the children of a `ParallaxLayer` a second time, so that it can look like it's wrapping around, and when the screen scrolls far enough - *ZIP!* - it whips back like a typewriter and makes it *appear* like it's scrolling forever!
```
Come on, it's not a stone tablet and chisel!

File: collections/_article/parallax-progress-report.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+After consulting with a few other contributors, it was clear there were two options:
+1. Wait until 5.0 and make these changes directly to `ParallaxLayer` and retire `ParallaxBackground`.
+2. Make a new node and provide a conversion tool (spoiler: we picked #2).
+
+It's risky either way. With the second option, having duplicate features could be confusing to users, but because because 4.0 only came out a year ago, users may be waiting a long time for these fixes if we push it out to a major release. Though... an entirely new node means we can make *even more* changes. >:)
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
It's risky either way. With the second option, having duplicate features could be confusing to users, but because 4.0 only came out a year ago, users may be waiting a long time for these fixes if we push it out to a major release. Though... an entirely new node means we can make *even more* changes. >:)
```

File: collections/_article/parallax-progress-report.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+#### What if I *want* it to work like a `CanvasLayer`?
+
+Here's the cool thing: *It still can!* Just put it in a `CanvasLayer`! The difference now is that you can choose. Just like any normal `Node2D`, placing a `Parallax2D` in a `CanvasLayer` will prevent zoom and rotating with the camera. You'll just want to make sure the `Parallax2D`'s `follow_viewport` value matches the `CanvasLayer`'s `follow_viewport_enabled` property. At this point you may notice that `Parallax2D`'s `follow_viewport` is enabled by default. That's because it *is* following the viewport! It's the core concept of the whole effect! This was a decision made because `ParallaxBackground`'s `follow_viewport_enabled` is *not* enabled even though it *is* following the viewport. It's confusing for a `CanvasLayer` to not behave like a `CanvasLayer`, so `Parallax2D` behaves more like one would expect.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
Here's the cool thing: *it still can!* Just put it in a `CanvasLayer`! The difference now is that you can choose. Just like any normal `Node2D`, placing a `Parallax2D` in a `CanvasLayer` will prevent zoom and rotating with the camera. You'll just want to make sure the `Parallax2D`'s `follow_viewport` value matches the `CanvasLayer`'s `follow_viewport_enabled` property. At this point you may notice that `Parallax2D`'s `follow_viewport` is enabled by default. That's because it *is* following the viewport! It's the core concept of the whole effect! This was a decision made because `ParallaxBackground`'s `follow_viewport_enabled` is *not* enabled even though it *is* following the viewport. It's confusing for a `CanvasLayer` to not behave like a `CanvasLayer`, so `Parallax2D` behaves more like one would expect.
```

File: collections/_article/parallax-progress-report.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+#### The right time to scroll to me
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]: ?

File: collections/_article/parallax-progress-report.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+`autoscroll` scrolls the layer *automatically* in addition to the parallax settings. Honestly, that's it! This could be done with a custom script, but it was highly requested, easy to implement, and had no performance hit if you weren't using it. So, why not?
+
+<video autoplay loop muted playsinline title="An example of parallax scrolling with autoscroll enabled">
+  <source src="/storage/blog/parallax2d/autoscroll.mp4" type="video/mp4">
+</video>
+
+#### Positive growth
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]: ?

File: collections/_article/parallax-progress-report.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  <source src="/storage/blog/parallax2d/zoom_out.mp4" type="video/mp4">
+</video>
+
+Increasing the `repeat_times` will allow for the number of repeats to grow and spread outward, so that you can zoom out and not reveal the magic going on behind the scenes. Eventually, it'd be nice to have a flag to have this adjust automatically, but this is a good start!
+
+#### What else?
+
+`Parallax2D` is now compatible with `BackBufferCopy` and `CanvasGroup`, and you're also free to rotate the camera without breaking the parallax effect! Some things needed shifted around in the `ParallaxBackground` code to accommodate the new changes and I discovered a nice surprise. Through some basic tests with 20 complex layers (comparing before and after the feature was added), I saw a good performance boost with both the old system and the new one with the changes... which is nice!
+
+### Caveats and Future
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
### Caveats and future
```

File: collections/_article/parallax-progress-report.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+With any effect that is based on perception, there are some tricky areas (since the entire effect is a trick!). A few of the same drawbacks to `ParallaxBackground` still apply to `Parallax2D`. For example, multiple cameras aren't directly supported (boooo!). This makes sense because `Parallax2D` follows the same concept as `ParallaxBackground`: Textures move at different speeds relative to the viewport. If you have multiple viewports, the textures can't move relative to all of them and can't be in two places at once! Just like in `ParallaxBackground`, as a workaround, you can clone the `Parallax2D`s and place each in a separate viewport. Besides the other future improvements mentioned earlier in this article, I'm also looking for ideas to better visualize parallax in the editor, whether that comes in the form of guides or another editor tool.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
With any effect that is based on perception, there are some tricky areas (since the entire effect is a trick!). A few of the same drawbacks to `ParallaxBackground` still apply to `Parallax2D`. For example, multiple cameras aren't directly supported (boooo!). This makes sense because `Parallax2D` follows the same concept as `ParallaxBackground`: textures move at different speeds relative to the viewport. If you have multiple viewports, the textures can't move relative to all of them and can't be in two places at once! Just like in `ParallaxBackground`, as a workaround, you can clone the `Parallax2D`s and place each in a separate viewport. Besides the other future improvements mentioned earlier in this article, I'm also looking for ideas to better visualize parallax in the editor, whether that comes in the form of guides or another editor tool.
```

File: collections/_article/parallax-progress-report.md (hunk 1 above)
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
`Parallax2D` is now compatible with `BackBufferCopy` and `CanvasGroup`, and you're also free to rotate the camera without breaking the parallax effect! Some things needed shifting around in the `ParallaxBackground` code to accommodate the new changes and I discovered a nice surprise. Through some basic tests with 20 complex layers (comparing before and after the feature was added), I saw a good performance boost with both the old system and the new one with the changes... which is nice!
```

File: collections/_article/parallax-progress-report.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+### Thanks!
+
+Thanks to those who helped make this feature. I would like to especially thank: [MewPurPur](https://github.com/MewPurPur) for their lightning-fast svg skills, [Mickeon](https://github.com/Mickeon) for championing my PR from the get-go, [KoBeWi](https://github.com/KoBeWi), [AThousandShips](https://github.com/AThousandShips), and [adamscott](https://github.com/adamscott) for jumping in with detailed reviews and constant questions to make sure it covered all corner cases and was *just right*... not to mention a dozen others that helped with feedback and testing.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
Thanks to those who helped make this feature. I would like to especially thank: [MewPurPur](https://github.com/MewPurPur) for their lightning-fast SVG skills, [Mickeon](https://github.com/Mickeon) for championing my PR from the get-go, [KoBeWi](https://github.com/KoBeWi), [AThousandShips](https://github.com/AThousandShips), and [adamscott](https://github.com/adamscott) for jumping in with detailed reviews and constant questions to make sure it covered all corner cases and was *just right*... not to mention a dozen others that helped with feedback and testing.
```

File: collections/_article/parallax-progress-report.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+I'm [Mark](https://github.com/markdibarry) (a.k.a. Mr. Dink in the Godot community). I've been contributing to the Godot engine for the last few years, but this is my first "big feature" contribution. If you have any suggestions for improvement for `Parallax2D`, please reach out or start a discussion. Heck! If you've got a good idea or see room for improvement and are hesitant to start contributing, don't be! Even if you see a misspelled word in the docs or a something you think might be too small, it's not! The thing that makes Godot unique is its community always willing to make it better.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
I'm [Mark](https://github.com/markdibarry) (a.k.a. Mr. Dink in the Godot community). I've been contributing to the Godot engine for the last few years, but this is my first "big feature" contribution. If you have any suggestions for improvement for `Parallax2D`, please reach out or start a discussion. Heck! If you've got a good idea or see room for improvement and are hesitant to start contributing, don't be! Even if you see a misspelled word in the docs or a something you think might be too small, it's not! The thing that makes Godot unique is its community is always willing to make it better.
```
Although tbh, I think it's exaggerating to say this is unique.

File: collections/_article/parallax-progress-report.md (hunk 2 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: I guess if I had to break it down, "Bogus" was a common phrase in the 90's meaning "disappointing", but is just used here to comedic effect (since it sounds very dated). It's just my writing style, but I can revise the post to remove things that don't convey a more serious tone.

If I had to pin-point the era for lighting in "early video game consoles", I'd say it'd be the 3rd (NES) through the 4th (SNES, Genesis, Gameboy). Before the 3rd, it was difficult to display much detail, and from somewhere between the 5th and 6th on they were able to use real lighting effects.

File: collections/_article/parallax-progress-report.md (hunk 4 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: Just another attempt at a fun sidebar. "Best in class" is a common award. `ParallaxLayer` is a class. I can remove it.

File: collections/_article/parallax-progress-report.md (hunk 5 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: Another joke; I'll remove it.

File: collections/_article/parallax-progress-report.md (hunk 2 above)
Comment (JohnVeness) [reply collections/_article/parallax-progress-report.md]: > I can revise the post to remove things that don't convey a more serious tone.

I'm not a super fan of the style, but I think it's OK to leave most of them in! I just dislike the ones that don't make sense. I'd be happy with "Disappointedly".

I think it would be better to say "NES-era" consoles, rather than "early". Early to me is Atari 2600!

File: collections/_article/parallax-progress-report.md (hunk 8 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: I was trying to think of flavorful section titles, and the song "Roll to Me" came to mind. I'll think of something else.

File: collections/_article/parallax-progress-report.md (hunk 9 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: The `repeat_times` field makes the textures grow... "positive growth" can also mean improvement. I'll think of another phrase.

File: collections/_article/parallax-progress-report.md (hunk 4 above)
Comment (JohnVeness) [reply collections/_article/parallax-progress-report.md]: Oh right, I get it now! Yeah, that didn't land with me!

File: collections/_article/parallax-progress-report.md (hunk 5 above)
Comment (JohnVeness) [reply collections/_article/parallax-progress-report.md]: I'm sure I'm coming over as a terribly boring person!

File: collections/_article/parallax-progress-report.md (hunk 2 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: It's okay - my style's not for everyone! Not a big deal.

I have seen it in the 3rd gen, but it was definitely more common in the 4th. I'll split the difference. Thanks for the feedback!

File: collections/_article/parallax-progress-report.md (hunk 5 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: No it's fine! I wasn't sure the tone I was supposed to be setting so I just went with what came naturally. I'm actually going to take another pass at the post in full. I think I was a little too playful with a lot of the phrasing.

File: collections/_article/parallax-progress-report.md (hunk 13 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: You're right. I removed that sentence.

File: collections/_article/parallax-progress-report.md (hunk 12 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: I removed the specific reasons for the shout outs. It was more playful than informative.

File: collections/_article/parallax-progress-report.md (hunk 2 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: Actually, now that you mention it, should I just remove references to my personal project? It's not 100% necessary to get the concept across, and digresses from the blog post early on. If I remove from the "I recently ran into..." to the "Is that a problem?" section. It'd make the blog post a lot more concise and easy to follow. I can just give a short sentence or two on how it used to work internally.

Comment (JohnVeness) [reply collections/_article/parallax-progress-report.md]: To be honest, I did get a bit lost during the explanation of your project and the problems you found. It might have been clearer with screenshots, but that would make the article even longer. If you can manage without the digression (while still trying to make it clear why the old system had limitations), I think it would indeed improve the article.

**Comment by markdibarry** [comment]
@JohnVeness I removed the informal tone, the references to my personal project, and unnecessary examples. Sorry again and let me know if there's any grammar mistakes. Thanks again for the feedback!

**Comment by JohnVeness** [comment]
> @JohnVeness I removed the informal tone, the references to my personal project, and unnecessary examples. Sorry again and let me know if there's any grammar mistakes. Thanks again for the feedback!

I think the grammar is (mostly) fine, but I think it's a shame to lose some of the videos. In particular, scroll.mp4 was good near the top of the article, and zoom_out.mp4 was helpful too.

File: collections/_article/parallax-progress-report.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+![An example of converting to Parallax2D](/storage/blog/parallax2d/button.webp)
+
+### Reasoning
+
+There have been a lot of proposals and issues regarding Godot's parallax system over the last few years. These had trouble getting movement without a significant overhaul of the two nodes that it's comprised of: `ParallaxBackground` and `ParallaxLayer`. Godot has introduced so many new features and upgrades in related areas that the parallax system didn't originally need to support.
+
+Originally, `ParallaxBackground` was intentionally made a `CanvasLayer` to take advantage of a quirk of the rendering system: Each layer is processed one at a time. This provides space to repeat entire branches multiple times, achieving `ParallaxLayer`'s "mirroring" effect. Not ideally, this setup is inflexible. We want to allow a parallax effect anywhere in the tree, not just at the root of a layer. By passing the `mirror` value further down in the renderer, we can have the same effect on a sub-branch. This also reveals something else significant: without the need for `ParallaxBackground` to be a `CanvasLayer`, there's not much of a need for `ParallaxBackground` at all.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
Originally, `ParallaxBackground` was intentionally made a `CanvasLayer` to take advantage of a quirk of the rendering system: each layer is processed one at a time. This provides space to repeat entire branches multiple times, achieving `ParallaxLayer`'s "mirroring" effect. Not ideally, this setup is inflexible. We want to allow a parallax effect anywhere in the tree, not just at the root of a layer. By passing the `mirror` value further down in the renderer, we can have the same effect on a sub-branch. This also reveals something else significant: without the need for `ParallaxBackground` to be a `CanvasLayer`, there's not much of a need for `ParallaxBackground` at all.
```

File: collections/_article/parallax-progress-report.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+After consulting with a few other contributors, it was clear there were two options:
+1. Wait until 5.0 and make these changes directly to `ParallaxLayer` and retire `ParallaxBackground`.
+2. Make a new node and provide a conversion tool.
+
+With the second option, having duplicate features could be confusing to users, but because 4.0 only came out a year ago, users may be waiting a long time for these fixes if we push it out to a major release.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]: I think (just to make it clear), you should add something like "We decided to go with the second option".

File: collections/_article/parallax-progress-report.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+With the big breaking change being up front (inheriting `Node2D`, and consolidating the nodes), it'll be much easier to add features or fixes without breaking changes in the future. Here is a list of some of the most common issues and requested features users had:
+
+- Ability to follow camera rotation
+- Compatibility with `CanvasGroup` and `BackBufferCopy`
+- Support for zooming out with a `Camera2D`
+- A simpler workflow
+- Autoscroll
+- Performance improvements
+- Better documentation
+
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md:61]: I think you should make it clear that this is not just a wishlist, but actually that Parallax2D has implemented these (as you had previously).

File: collections/_article/parallax-progress-report.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+#### `CanvasLayer` to `Node2D`
+
+You can still get the benefits of a `CanvasLayer`. The difference now is that you can choose. Just like any normal `Node2D`, placing a `Parallax2D` in a `CanvasLayer` will prevent zoom and rotating with the camera. You'll just want to make sure the `Parallax2D`'s `follow_viewport` value matches the `CanvasLayer`'s `follow_viewport_enabled` property. At this point you may notice that `Parallax2D`'s `follow_viewport` is enabled by default. This was a decision made because `ParallaxBackground`'s `follow_viewport_enabled` is not enabled even though it is following the viewport. It's confusing for a `CanvasLayer` to not behave like a `CanvasLayer`, so `Parallax2D` behaves more like one would expect.
+
+#### `autoscroll`
+
+`autoscroll` scrolls the layer automatically in addition to the parallax settings. This could be done with a custom script, but decided to do so anyway since it was highly requested, easy to implement, and had no performance hit if you weren't using it.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
`autoscroll` scrolls the layer automatically in addition to the parallax settings. This could be done with a custom script, but we decided to put this into `Parallax2D` since it was highly requested, easy to implement, and has no performance hit if you aren't using it.
```

**Comment by markdibarry** [comment]
Gotcha! I added those two examples back and made the requested changes. Let me know if you see anything else!

File: collections/_article/parallax-progress-report.md [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+
+### Thank you
+
+Thanks to those who helped make this feature, especially: [MewPurPur](https://github.com/MewPurPur), [Mickeon](https://github.com/Mickeon), [KoBeWi](https://github.com/KoBeWi), [AThousandShips](https://github.com/AThousandShips), and [adamscott](https://github.com/adamscott), not to mention a dozen others that helped with feedback and testing. If you have any suggestions for improvement for `Parallax2D`, please make a proposal or start a discussion. If you've got a good idea or see room for improvement and are hesitant to start contributing, don't be! Even if you see a misspelled word in the docs or a something you think might be too small, it's not!
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]: Add yourself to this list, Mark!

Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md:84]: Add yourself to this list, Mark!

Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: Haha. Totally forgot! Will do!

**Comment by markdibarry** [comment]
Looks like there was some more confusion on my part! Sorry again! After some clarification, it looks like I *am* supposed to be the author of the blog post after all. I've added some of the original tone back, but scaled back a lot of the post since it was found to be hard to follow. I updated the zooming video to be easier to follow. I also removed the references to songs by Del Amitri, but I'm putting my foot down on the word "bogus". :joy:

It's getting late my time but I'll do another look over tomorrow morning!

File: collections/_article/parallax-progress-report.md [hunk 19]
```diff
@@ -15,27 +15,29 @@ Simulating depth with backgrounds is a staple of 2D games. Godot's parallax syst
...
+`Parallax2D` is a new node intended as an all-in-one replacement for `ParallaxBackground` and `ParallaxLayer`. It has feature-parity with the old system along with some new goodies! This is only the beginning, though; There's tons of improvements that can still be made! It's flagged as experimental in 4.3 and is subject to change while we get some more feedback. The plan is to eventually deprecate (not remove) the old nodes, but rest assured, they won't be deprecated until we are confident they are fully superseded by `Parallax2D`. New features will be focused on `Parallax2D`, but we will still be accepting bug fixes for the old nodes.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
`Parallax2D` is a new node intended as an all-in-one replacement for `ParallaxBackground` and `ParallaxLayer`. It has feature-parity with the old system along with some new goodies! This is only the beginning, though; there's tons of improvements that can still be made! It's flagged as experimental in 4.3 and is subject to change while we get some more feedback. The plan is to eventually deprecate (not remove) the old nodes, but rest assured, they won't be deprecated until we are confident they are fully superseded by `Parallax2D`. New features will be focused on `Parallax2D`, but we will still be accepting bug fixes for the old nodes.
```

File: collections/_article/parallax-progress-report.md [hunk 20]
```diff
@@ -95,8 +97,12 @@ Increasing the `repeat_times` will allow for the number of repeats to grow and s
 
 With any effect that is based on perception, there are some tricky areas (since the entire effect is a trick!). A few of the same drawbacks to `ParallaxBackground` still apply to `Parallax2D`. For example, multiple cameras aren't directly supported (booo!). This makes sense because `Parallax2D` follows the same concept as `ParallaxBackground`: textures move at different speeds relative to the viewport. If you have multiple viewports, the textures can't move relative to all of them and can't be in two places at once! Just like in `ParallaxBackground`, as a workaround, you can clone the `Parallax2D`s and place each in a separate viewport.
 
+Additionally, even though `Parallax2D` is a `Node2D`, you should still take caution when moving or scaling any parent nodes. It can be a delicate effect, so its recommended to be careful in complex setups.
```
Comment (JohnVeness) [inline collections/_article/parallax-progress-report.md]:
```suggestion
Additionally, even though `Parallax2D` is a `Node2D`, you should still take caution when moving or scaling any parent nodes. It can be a delicate effect, so it's recommended to be careful in complex setups.
```

**Comment by coppolaemilio** [comment]
We can get this one ready for tomorrow! I'll review it today and see what's missing 👍

**Review by coppolaemilio — APPROVED** [review]
I love it! really good work 👏  loved every video as well, they are super cute :)

File: collections/_article/parallax-progress-report.md [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+
+After consulting with a few other contributors, it was clear there were two options:
+1. Wait until 5.0 and make these changes directly to `ParallaxLayer` and retire `ParallaxBackground`.
+2. Make a new node and provide a conversion tool.
+
+BIG SURPRISE: we decided to go with the second option! Having duplicate features could be confusing to users, but because 4.0 only came out a year ago, users may be waiting a long time for these fixes if we push it out to a major release. Though... an entirely new node means we can make *even more* changes!
```
Comment (MurderVeggie) [inline collections/_article/parallax-progress-report.md]: > BIG SURPRISE: we decided to go with the second option!

I really like your writing style, and don't mind that it leans more informal and humorous, if anything that makes it a pleasure to read in my book! However, here I would avoid going all caps since that is a little misleading in terms of importancy. I would remove this sentence and simply start the next one with "While..."

Additionally, I would rephrase the last one to "By creating an entirely new node we have the freedom to make these changes immediately, and *even more* of them"

File: collections/_article/parallax-progress-report.md [hunk 22]
```diff
@@ -0,0 +1,112 @@
...
+- Support for zooming out with a `Camera2D`
+- Autoscroll
+- Improved documentation
+- Performance improvements
+
+#### Reogranization
+
+![The Parallax2D inspector](/storage/blog/parallax2d/inspector.webp)
+
+The inspector properties are now reorganized and categorized. Notice that the limits now follow the same logic as `Camera2D` limits. How *standardized!* There are notes in the previous parallax system's documentation clarifying that `mirroring` doesn't actually mirror anything, but rather just repeats. The name has been changed to `repeat_size` to better reflect (ha!) what it actually does.
```
Comment (MurderVeggie) [inline collections/_article/parallax-progress-report.md]: > (ha!)

Not needed, I would remove this one.

File: collections/_article/parallax-progress-report.md [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+
+#### What else?
+
+`Parallax2D` is now compatible with `BackBufferCopy` and `CanvasGroup`, and you're also free to rotate the camera without breaking the parallax effect! Upon a few initial tests, we noticed some performance increases as well.
+
+### Caveats and future
+
+With any effect that is based on perception, there are some tricky areas (since the entire effect is a trick!). A few of the same drawbacks to `ParallaxBackground` still apply to `Parallax2D`. For example, multiple cameras aren't directly supported (booo!). This makes sense because `Parallax2D` follows the same concept as `ParallaxBackground`: the textures' positions are modified directly and move at different speeds relative to the viewport. If you have multiple viewports, the textures can't move relative to all of them and can't be in two places at once! Just like in `ParallaxBackground`, as a workaround, you can clone the `Parallax2D`s and place each in a separate viewport.
```
Comment (MurderVeggie) [inline collections/_article/parallax-progress-report.md]: > (booo!)

Again, the sentence leaves a better impression on the reader without the interjection taking away from it, so I would cull it.

File: collections/_article/parallax-progress-report.md [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+
+If you have any suggestions for improvement for `Parallax2D`, feel free to make a proposal or start a discussion. We already have some bug fixes on the way, so be sure to check the backlog first! If you've got a good idea or see room for improvement and are hesitant to start contributing, don't be! Even if you see a misspelled word in the docs or a something you think might be too small, it's not! We wouldn't be where we are without the willingness of our community to jump in and get their hands dirty... so join in!
```
Comment (MurderVeggie) [inline collections/_article/parallax-progress-report.md]: >  If you've got a good idea or see room for improvement and are hesitant to start contributing, don't be! Even if you see a misspelled word in the docs or a something you think might be too small, it's not! We wouldn't be where we are without the willingness of our community to jump in and get their hands dirty... so join in!

I really like ending on a call to contribute, but I would slightly rephrase it for more impact. My proposal:

`Do you see room for improvement or felt the spark of an idea while reading this? We are always looking for more contributors and would be happy to have you. Don't worry about the size of your first submission, many of us have started with rectifying spelling errors in the docs. We wouldn't be where we are without the willingness of our community to jump in and get their hands dirty. What are you waiting for?`

**Review by MurderVeggie — CHANGES_REQUESTED** [review]
Hey there! Had a quick read through focussing on the language side of things, since that is where my expertise lies. Hope my suggestions make sense to you, if not please ask about them. Overall I loved your blog post, can't wait to share it with the community!

File: collections/_article/parallax-progress-report.md (hunk 24 above)
Comment (markdibarry) [reply collections/_article/parallax-progress-report.md]: Love this!

**Review by MurderVeggie — APPROVED** [review]

**Comment by coppolaemilio** [comment]
Thanks and congrats on surviving the process of getting the blog post released @markdibarry 🥳

**Comment by MurderVeggie** [comment]
Seconding this, would love to see more from you in the future!
